lighthouse-core/lib/page-functions.js

```diff
@@ -467,26 +467,27 @@ function getNodeDetailsImpl(element) {
     window.__lighthouseNodesDontTouchOrAllVarianceGoesAway = new Map();
   }
 
+  const htmlElement = element instanceof ShadowRoot ? element.host : element;
+
   // Create an id that will be unique across all execution contexts.
   // The id could be any arbitrary string, the exact value is not important.
   // For example, tagName is added only because it might be useful for debugging.
   // But execution id and map size are added to ensure uniqueness.
   // We also dedupe this id so that details collected for an element within the same
   // pass and execution context will share the same id. Not technically important, but
   // cuts down on some duplication.
-  let lhId = window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.get(element);
+  let lhId = window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.get(htmlElement);
   if (!lhId) {
     lhId = [
       window.__lighthouseExecutionContextId !== undefined ?
         window.__lighthouseExecutionContextId :
         'page',
       window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.size,
-      element.tagName,
+      htmlElement.tagName,
     ].join('-');
-    window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.set(element, lhId);
+    window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.set(htmlElement, lhId);
```

Comment:
the other option is to do another ShadowRoot check here:

https://github.com/GoogleChrome/lighthouse/blob/966a206cb3c4fd2317788c956b6007064e7c33e5/lighthouse-core/gather/gatherers/full-page-screenshot.js#L88

but this seems in keeping with the intent of `getNodeDetails` (see the `getBoundingClientRect(htmlElement)`, below)